pkg/extra/do/mockcloud/client.go

```diff
@@ -830,3 +834,53 @@ func (mock *MockLoadBalancers) RemoveForwardingRules(ctx context.Context, lbId s
 
 	return mock.wrap.LoadBalancers().RemoveForwardingRules(ctx, lbId, rules...)
 }
+
+// Snapshots
+
+type MockSnapshots struct {
+	wrap          cloud.Client
+	GetFn         func(ctx context.Context, sId string) (snapshots.Snapshot, error)
+	DeleteFn      func(ctx context.Context, sId string) error
+	ListFn        func(ctx context.Context) (<-chan snapshots.Snapshot, <-chan error)
+	ListDropletFn func(ctx context.Context) (<-chan snapshots.Snapshot, <-chan error)
+	ListVolumeFn  func(ctx context.Context) (<-chan snapshots.Snapshot, <-chan error)
+}
+
+func (mock *MockSnapshots) Get(ctx context.Context, sId string) (snapshots.Snapshot, error) {
+	if mock.GetFn != nil {
+		return mock.GetFn(ctx, sId)
+	}
+
+	return mock.wrap.Snapshots().Get(ctx, sId)
+}
+
+func (mock *MockSnapshots) Delete(ctx context.Context, sId string) error {
+	if mock.GetFn != nil {
+		return mock.DeleteFn(ctx, sId)
+	}
+
+	return mock.wrap.Snapshots().Delete(ctx, sId)
+}
+
+func (mock *MockSnapshots) List(ctx context.Context) (<-chan snapshots.Snapshot, <-chan error) {
+	if mock.ListFn != nil {
+		return mock.ListFn(ctx)
+	}
+
+	return mock.wrap.Snapshots().List(ctx)
+}
+func (mock *MockSnapshots) ListDroplet(ctx context.Context) (<-chan snapshots.Snapshot, <-chan error) {
+	if mock.ListFn != nil {
```

Comment:
I figured out why your `TestSnapshotListDroplet` function is failing. You have a typo here. It should be `mock.ListDropletFn != nil`